langchain4j/src/main/java/dev/langchain4j/service/ServiceOutputParser.java

```diff
@@ -130,7 +130,11 @@ private static String jsonStructure(Class<?> structured, Set<Class<?>> visited)
             }
             jsonSchema.append(format("\"%s\": (%s),\n", name, descriptionFor(field, visited)));
         }
-        jsonSchema.delete(jsonSchema.lastIndexOf(","), jsonSchema.lastIndexOf(",")+1);
+
+        int trailingCommaIndex = jsonSchema.lastIndexOf(",");
+        if (trailingCommaIndex > 0) {
+            jsonSchema.delete(trailingCommaIndex, trailingCommaIndex +1);
+        }
         jsonSchema.append("}");
```

Comment:
Thanks for the fix!

I just ran into the same issue, and was considering fixing it with `String.join()` instead of the gymnastics around the corner cases.

Ie, the whole function would become something like:

```
private static String jsonStructure(Class<?> structured, Set<Class<?>> visited) {
        List<String> descriptions = new ArrayList<>();
        for (Field field : structured.getDeclaredFields()) {
            String name = field.getName();
            if (name.equals("__$hits$__") || java.lang.reflect.Modifier.isStatic(field.getModifiers())) {
                // Skip coverage instrumentation field.
                continue;
            }
            descriptions.add(format("\"%s\": (%s)", name, descriptionFor(field, visited)));
        }

        StringBuilder jsonSchema = new StringBuilder();
        jsonSchema.append("{\n");
        jsonSchema.append(String.join(",\n", descriptions));;
        jsonSchema.append("\n}");

        return jsonSchema.toString();
    }
```